docs/csharp/whats-new/csharp-6.md

```diff
@@ -98,483 +36,157 @@ public class Student
-
-After adding this in code, you set your debugger to break on all unhandled
-exceptions. Run the program under the debugger, and the debugger breaks
-whenever `PerformFailingOperation()` throws a `RecoverableException`.
-The debugger breaks your program, because the catch clause won't be executed
-due to the false-returning exception filter.
-
 ## The `nameof` expression
 
-The `nameof` expression evaluates to the name of a symbol. It's a great
-way to get tools working whenever you need the name of a variable,
-a property, or a member field.
-
-One of the most common uses for `nameof` is to provide the name of a symbol
-that caused an exception:
+The `nameof` expression evaluates to the name of a symbol. It's a great way to get tools working whenever you need the name of a variable, a property, or a member field. One of the most common uses for `nameof` is to provide the name of a symbol that caused an exception:
 
 [!code-csharp[nameof](../../../samples/snippets/csharp/new-in-6/NewCode.cs#UsingStaticString)]
 
-Another use is with XAML based applications that implement the `INotifyPropertyChanged`
-interface:
+Another use is with XAML-based applications that implement the `INotifyPropertyChanged` interface:
 
 [!code-csharp[nameofNotify](../../../samples/snippets/csharp/new-in-6/viewmodel.cs#nameofNotify)]
 
-The advantage of using the `nameof` operator over a constant string is
-that tools can understand the symbol. If you use refactoring tools to
-rename the symbol, it will rename it in the `nameof` expression. Constant
-strings don't have that advantage. Try it yourself in your favorite editor:
-rename a variable, and any `nameof` expressions will update as well.
-
-The `nameof` expression produces the unqualified name of its argument
-(`LastName` in the previous examples) even if you use the fully qualified
-name for the argument:
-
-[!code-csharp[QualifiedNameofNotify](../../../samples/snippets/csharp/new-in-6/viewmodel.cs#QualifiedNameofNotify)]
-
-This `nameof` expression produces `FirstName`, not `UXComponents.ViewModel.FirstName`.
-
 ## Await in Catch and Finally blocks
 
-C# 5 had several limitations around where you could place `await` expressions.
-One of those has been removed in C# 6. You can now use `await` in `catch`
-or `finally` expressions. 
-
-The addition of await expressions in catch and finally blocks may appear
-to complicate how those are processed. Let's add an example to discuss
-how this appears. In any async method, you can use an await expression
-in a finally clause.
-
-With C# 6, you can also await in catch expressions. This is most often
-used with logging scenarios:
+C# 5 had several limitations around where you could place `await` expressions. You can now use `await` in `catch` or `finally` expressions.  With C# 6, you can also await in catch expressions. This is most often used with logging scenarios:
```

Comment:
This is confusing. In C# 5, can you await in a finally, as this seems to imply? The mention of limitations in C# 5 also isn't particularly relevant, since it doesnt list the limitations. I've suggested some changes.